components/tikv_util/src/mpsc/batch.rs

```diff
@@ -422,33 +440,64 @@ mod tests {
 
     #[test]
     fn test_switch_between_sender_and_receiver() {
-        struct Notifier<F>(Arc<Mutex<Option<Spawn<F>>>>);
-        impl<F> Clone for Notifier<F> {
-            fn clone(&self) -> Notifier<F> {
-                Notifier(Arc::clone(&self.0))
-            }
+        let (task_sender, executor) = create_sender_and_executor();
```

Comment:
This is too complicated, I have reviewed it for more than 10 minutes. Can it be simplified? The extra `task_sender` and `executor` are distracting.